pkg/network/node/ovscontroller.go

```diff
@@ -686,6 +692,9 @@ func (oc *ovsController) UpdateNamespaceEgressRules(vnid uint32, nodeIP, egressH
 		otx.AddFlow("table=100, priority=100, reg0=%d, actions=drop", vnid)
 	} else if nodeIP == oc.localIP {
 		// Local Egress IP
+		if err := oc.ensureTunMAC(); err != nil {
```

Comment:
Now I'm thinking if tunMAC is specific to egressIP then why to keep it as part of ovsController?
This may be clean:
- No 'tunMAC' variable in ovsController but can can have 'GetTunMAC' that return tun0 mac address.
- Introduce 'localEgressMAC' variable in egressIPWatcher struct.
- Set localEgressMAC = oc.GetTunMAC in eip.Start() [Where localEgressLink and localEgressNet are also set]
- Pass localEgressMAC to oc.UpdateNamespaceEgressRules()

Reply:
> Now I'm thinking if 'tunMAC' is specific to egress IP then why to keep it as part of ovsController?

It's only used by the egressIP-related code, but the need for it is specific to ovsController.

eg, contrast with the egressHex value; there we need to have the same value in the OVS flow as in the iptables rule, so the egress IP code computes that and passes it to UpdateNamespaceEgressRules() in ovscontroller.go and AddEgressIPRules() in iptables.go.

But for tunMAC, the fact that we need it at all is just a side effect of other decisions inside ovscontroller. If the ovscontroller did other things differently, then we wouldn't need to refer to tun0's MAC address here. So I think keeping it in ovscontroller is correct.

Reply:
okay